Source/missiles.cpp

```diff
@@ -861,48 +895,24 @@ bool PlayerMHit(int pnum, int m, int dist, int mind, int maxd, int mtype, bool s
 			return true;
 		}
 
-		if (mtype == MIS_BONESPIRIT) {
-			dam = plr[pnum]._pHitPoints / 3;
-		} else {
-			if (!shift) {
-
-				dam = (mind << 6) + GenerateRnd((maxd - mind + 1) << 6);
-				if (m == -1)
-					if ((plr[pnum]._pIFlags & ISPL_ABSHALFTRAP) != 0)
-						dam /= 2;
-				dam += (plr[pnum]._pIGetHit << 6);
-			} else {
-				dam = mind + GenerateRnd(maxd - mind + 1);
-				if (m == -1)
-					if ((plr[pnum]._pIFlags & ISPL_ABSHALFTRAP) != 0)
-						dam /= 2;
-				dam += plr[pnum]._pIGetHit;

```

Comment:
This part was bothering me. Sometimes `_pIGetHit` is shifted, sometimes it isn't. The same flag that determines whether a missile collision is blockable also determines whether this value (and the min/max damage passed in) should be shifted by this logic. Missiles that do not get shifted include acid puddles, firewall, lightning wall, lightning, flash, inferno, and the explosion from Elemental.

I assumed that the value stored in `_pIGetHit` is normally unshifted, and so the unshifted value would do a negligible amount of damage (1/64th of the normal damage). Because of that, I did not include the unshifted adjustment in the cleaned up logic. However, I do also notice that many of these missiles are multi-hit missiles so the small damage adjustment could add up. Still, I believe that few of these damage sources will hit you 64 times before you either move away, stop getting hit, or die.

If anyone has an opinion, I would appreciate some input.

Reply:
In thinking about it a little bit more, if you have Stormshield's +4 Damage from Enemies, an acid puddle or firewall could end up doing +80 damage per second with the shifted value, which seems problematic. I wonder if maybe these attacks should apply the damage adjustment per second. Or in the case of lightning and inferno, use the duration of the missile as the divisor.

Reply:
I put the unshifted adjustment back in for now. I still think it's wrong to arbitrarily apply 1/64th of the damage adjustment to all these missiles. In particular, it makes no sense for the explosion from Elemental. Please chime in if you have any thoughts.